src/Processors/Formats/Impl/BinaryRowInputFormat.cpp

```diff
@@ -23,10 +23,44 @@ BinaryRowInputFormat<with_defaults>::BinaryRowInputFormat(ReadBuffer & in_, cons
         with_names_,
         with_types_,
         format_settings_,
-        std::make_unique<BinaryFormatReader<with_defaults>>(in_, format_settings_))
+        std::make_unique<BinaryFormatReader<with_defaults>>(in_, format_settings_)), binary_reader(assert_cast<BinaryFormatReader<with_defaults> *>(format_reader.get()))
 {
 }
 
+template <bool with_defaults>
+bool BinaryRowInputFormat<with_defaults>::readRow(DB::MutableColumns & columns, DB::RowReadExtension & ext)
```

Comment:
`RowInputFormatWithNamesAndTypes::readRow` makes some virtual calls and checks that are unnecessary for `RowBinary` format.